src/test/java/plugins/JavadocPluginTest.java

```diff
@@ -35,11 +39,30 @@ public void publish_javadoc_from_matrix_job() {
 
         job.startBuild().shouldSucceed();
         MatrixConfiguration def = job.getConfiguration("default");
-        assertThat(def, hasAction("Javadoc"));
+        assertThat(def, hasAction(JAVADOC_ACTION));
 
         assertJavadoc(def);
     }
 
+    @Test
+    public void validate_javadoc_retention() {
+        FreeStyleJob job = jenkins.jobs.create();
+        setup(job);
+        Build b = job.startBuild().shouldSucceed();
+
+        assertThat(job, hasAction(JAVADOC_ACTION));
+        try {
+            assertThat(b, hasAction(JAVADOC_ACTION));
+            Assert.fail("Build #1 should not have Javadoc action");
+        } catch (AssertionError ex) {}
```

Comment:
Suggestion. Could this whole try block simply be replaced for the following? 

`assertThat("Build #1 should not have Javadoc action", b, not(hasAction(JAVADOC_ACTION)))`